contrib/lua/premake5.lua

```diff
@@ -21,8 +21,8 @@ project "lua-lib"
 		"etc/*.c"
 	}
 
-	configuration "linux or bsd or hurd or aix"
+	filter "system:linux or system:bsd or system:hurd or system:aix"
```

Comment:
Thtat's why I didn't do it.

And yeah the examples only show kind: being used like you described, but it's not explicitly mentioned like the other stuff that's happening, unless  

> If any keyword pattern fails to match the current context, the entire filter is skipped over.

is supposed to be that hint...

Sometimes I miss a talk page for each wiki page like with mediawiki...

PS: seems like my intuitive "not" thinking is wrong, do you really need to put "not" between the scope and the keyword? (after looking at the code, you apparently need to, still I think that you shouldn't separate the keyword from it's scope trough a not, but that's just my opinion...)

Actually I just looked the filter code up and noticed that [if there's no scope then scope "configurations" is assumed](https://github.com/premake/premake-core/blob/master/src/base/criteria.lua#L66), should I reflect this in the PR, unless this detail is discouraged like the comments mention.

Also seems like "iif" isn't documented in the wiki (it's not even in the "lua additions" list), did it go missing in the transition between bitbucket and github?